src/components/shared/Modal.tsx

```diff
@@ -62,94 +62,111 @@ export function Modal({ closeModal, patientData, isEditMode }: ModalProps) {
   };
 
   return (
-    <fieldset className="fixed flex justify-center items-center w-full h-screen top-0 left-0 bg-[rgba(0,0,0,0.4)]">
-      <fieldset className="bg-white p-6 w-[30rem] h-[35rem] rounded-md">
-        <div className="flex justify-between items-center mb-4">
-          <p className="text-xl font-bold">
+    <fieldset className="fixed inset-0 flex justify-center items-center bg-black bg-opacity-50 backdrop-blur-sm">
+      <fieldset className="bg-white p-8 w-full max-w-md rounded-lg shadow-xl transform transition-all duration-300">
+        <div className="flex justify-between items-center mb-6">
+          <p className="text-2xl font-semibold text-gray-700">
             {isEditMode ? "Edit Patient" : "Add New Patient"}
           </p>
-          <button className="text-xl font-bold" onClick={closeModal}>
-            x
+          <button
+            className="text-gray-400 hover:text-gray-600 transition duration-200"
+            onClick={closeModal}
+          >
+            &#x2715;
```

Comment:
A co to za piękny znacznik :D musimy zrobić tak, aby urządzenia a11y asystujące odczytywały inną wartość dla użtkownika z niepełnosprawnością. aria-label może w tym pomóc